src/test/kotlin/graphics/scenery/tests/examples/advanced/PersistentTextureRequestsExample.kt

```diff
@@ -0,0 +1,126 @@
+ * textures need to be accessed every time they are updated.
+ *
+ * @author Aryaman Gupta <argupta@mpi-cbg.de>
+ * @author Ulrik Günther <hello@ulrik.is>
+ */
+class PersistentTextureRequestsExample : SceneryBase("PersistentTextureRequestsExample") {
+
+    private val counter = AtomicInteger(0)
+    private var totalFrames = -1L
+
+    override fun init() {
+        renderer = hub.add(
+            SceneryElement.Renderer,
+            Renderer.createRenderer(hub, applicationName, scene, 1280, 720))
+
+        val volumeManager = VolumeManager(hub,
+            useCompute = true,
+            customSegments = hashMapOf(
+                SegmentType.FragmentShader to SegmentTemplate(
+                    this.javaClass,
+                    "ComputeVolume.comp",
+                    "intersectBoundingBox", "vis", "SampleVolume", "Convert", "Accumulate"),
+            ))
+
+        val outputBuffer = MemoryUtil.memCalloc(1280*720*4)
+        val outputTexture = Texture.fromImage(Image(outputBuffer, 1280, 720), usage = hashSetOf(Texture.UsageType.LoadStoreImage, Texture.UsageType.Texture))
+        volumeManager.material.textures["OutputRender"] = outputTexture
+
+        hub.add(volumeManager)
+
+        val imp: ImagePlus = IJ.openImage("https://imagej.nih.gov/ij/images/t1-head.zip")
+        val img: Img<UnsignedShortType> = ImageJFunctions.wrapShort(imp)
+
+        val volume = Volume.fromRAI(img, UnsignedShortType(), AxisOrder.DEFAULT, "T1 head", hub, VolumeViewerOptions())
+        volume.transferFunction = TransferFunction.ramp(0.001f, 0.5f, 0.3f)
+        scene.addChild(volume)
+
+        val box = Box(Vector3f(1.0f, 1.0f, 1.0f))
+        box.name = "le box du win"
+        box.material.textures["diffuse"] = outputTexture
+        box.material.metallic = 0.0f
+        box.material.roughness = 1.0f
+
+        scene.addChild(box)
+
+        val light = PointLight(radius = 15.0f)
+        light.position = Vector3f(0.0f, 0.0f, 2.0f)
+        light.intensity = 5.0f
+        light.emissionColor = Vector3f(1.0f, 1.0f, 1.0f)
+        scene.addChild(light)
+
+        val cam: Camera = DetachedHeadCamera()
+        with(cam) {
+            position = Vector3f(0.0f, 0.0f, 5.0f)
+            perspectiveCamera(50.0f, 512, 512)
+
+            scene.addChild(this)
+        }
+
+        thread {
+            val opTexture = volumeManager.material.textures["OutputRender"]!!
+
+            val prevCounter = 0
+
+            renderer?.persistentTextureRequests?.add(opTexture to counter)
+
+            // this is the loop where you may do your tasks that are asynchronous to rendering
+            while (renderer?.shouldClose == false) {
+                while (counter.get() == prevCounter) {
+                    Thread.sleep(5)
+                }
+
+                //updated texture has been returned
+                val buffer = opTexture.contents
+                //the buffer can now, e.g., be transmitted, as is required for parallel rendering
```

Comment:
Maybe use the buffer as a texture for another object here, for demo purposes?

Reply:
A better example would be to write the texture to disk. Will that create any problems in the CI?
Using it as a texture in another object is not such a great example because you can do that without fetching the texture from the GPU